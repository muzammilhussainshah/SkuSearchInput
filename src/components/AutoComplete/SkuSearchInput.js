import React, { useState, useEffect } from "react";
import AutoCompleteComponent from "../AutoComplete/index";

export default function SkuSearchInput({
  searchUrl,
  checkSearchKey,
  width,
  placeHolder,
  backgroundColor,
  onSelect,
}) {
  const [autoFilledData, setautoFilledData] = useState([]);
  const [isLoading, setisLoading] = useState(false);
  const [searchText, setSearchText] = useState("");
  const [noResults, setNoResults] = useState(false);

  const searchData = async (keyWord) => {
    setisLoading(true);
    let url;
    url = `${searchUrl}${checkSearchKey}=${keyWord}`;
    try {
      var requestOptions = {
        method: "GET",
        redirect: "follow",
      };
      fetch(url, requestOptions)
        .then((response) => response.json())
        .then((result) => {
          console.log(
            "ðŸš€ ~ file: SkuSearchInput.js ~ line 32 ~ .then ~ result",
            result
          );
          setautoFilledData(result.sku);
          setisLoading(false);
        })
        .catch((error) => console.log("error", error));
    } catch (err) {
      setisLoading(false);
      console.log(
        err.response,
        "error from searchData",
        JSON.parse(JSON.stringify(err.message))
      );
    }
  };

  let timer = setTimeout(() => { }, 1);
  const onChange = (keyWord) => {
    if (keyWord) {
      clearTimeout(timer);
      timer = setTimeout(() => {
        setSearchText(keyWord);
        searchData(keyWord);
      }, 500);
    } else {
      setTimeout(() => {
        setautoFilledData([]);
      }, 1200);
    }
  };

  return (
    <div style={{ width }}>
      <AutoCompleteComponent
        loading={isLoading}
        data={autoFilledData}
        onFocus={() => {
          setSearchText(true);
          setNoResults(true);
        }}
        onBlur={() => {
          setTimeout(() => {
            setSearchText(false);
            setNoResults(false);
          }, 500);
        }}
        onChangeText={(value) => onChange(value)}
        placeHolder={placeHolder}
        backgroundColor={backgroundColor}
        noResults={noResults}
        searchText={searchText}
        onSelect={(value) => onSelect(value)}
      />
    </div>
  );
}
