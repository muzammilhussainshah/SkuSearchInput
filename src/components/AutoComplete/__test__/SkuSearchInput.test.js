import React from 'react';
import { render, cleanup, } from '@testing-library/react';
import SkuSearchInput from '../SkuSearchInput';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';


afterEach(cleanup)
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<SkuSearchInput ></SkuSearchInput >, div);
})


it('renders SkuSearchInput correctly', () => {
  const { getByTestId, } =
    render(
      <SkuSearchInput
        searchUrl={'https://toro278.us-east.toroserver.com/api/demo_api_inventory/1.0/sku/search?'}
      ></SkuSearchInput >
    )
  expect(getByTestId("search-button")).toBeTruthy();
})

it('matches snapshot', () => {
  const tree = renderer.create(<SkuSearchInput
    searchUrl={'https://toro278.us-east.toroserver.com/api/demo_api_inventory/1.0/sku/search?'}
  ></SkuSearchInput >).toJSON();
  expect(tree).toMatchSnapshot();
})

