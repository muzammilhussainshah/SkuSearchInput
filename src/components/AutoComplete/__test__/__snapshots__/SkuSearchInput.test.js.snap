// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="search-button"
  style={
    Object {
      "width": undefined,
    }
  }
>
  <div
    className="search-input"
  >
    <div
      className="input-container"
      style={
        Object {
          "backgroundColor": undefined,
          "borderColor": undefined,
          "borderStyle": "solid",
          "borderWidth": 1,
        }
      }
    >
      <input
        className="input-autocomplete"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        value=""
      />
      <div>
        <img
          src="search.png"
          width={30}
        />
      </div>
    </div>
    
    
  </div>
</div>
`;
